### PR TITLE
Blockly Signal Block

### PR DESCRIPTION
**Pull request by AmbientOSX**
@kadst43 Also made some various changes to arithmetic blocks and number blocks that I assumed we'd want.

### CODE REVIEW

File: source/rover.js [hunk 1]
```diff
@@ -314,6 +323,29 @@ this.activateSensor = function( sensor ) {
...
+
+        var deltaX = signalPos[ 0 ] - currentPos[ 0 ];
+        var deltaY = signalPos[ 1 ] - currentPos[ 1 ];
+
+        var radians = Math.atan2( deltaY, deltaX ); // In radians 
+        var heading = radians * ( 180 / Math.PI );
+
+        // Convert to 0 to 360 
+        if ( heading >= 0 ) {
+            heading = 180 - heading;
```
Comment (kadst43) [inline source/rover.js]: What does this line do?  It seems wrong...

File: source/view/blocks.js [hunk 2]
```diff
@@ -644,13 +645,68 @@ Blockly.Blocks[ 'controls_sensor_tracks' ] = {
...
+    this.setColour( 30 );
+    this.appendValueInput('INPUT')
+        .appendField('Base Signal °')
+        .setCheck(['OperatorAddSubtract','OperatorMultiplyDivide','LeftParenthesis','RightParenthesis','Conditional']);
+    this.setOutput(true, null);
+
+    var thisBlock = this;
+    this.setTooltip( function() {
+      var content = {
+        text: "Checks our scanner for base location in a 360° arc that starts on the X-axis. Not yet available on Perry!"
```
Comment (kadst43) [inline source/view/blocks.js]: Remove the "Not yet available on Perry!" part.

Also, on text like this let's call him Peregrine.  Perry is a nickname (that he hates) - I think that outside of us developers, Rosie may be the only one who uses it.

**Comment by kadst43** [comment]
To test... set the signal in each of the four quadrants, then write a program that will take you to it using this block.

File: source/view/blocks.js (hunk 2 above)
Comment (kadst43) [inline source/view/blocks.js]: Also, I might specify that it returns a value between 0 and 360 (or -180 and 180, or whatever).

**Comment by kadst43** [comment]
After testing, and updating the tooltip, :+1:

**Comment by kadst43** [comment]
@AmbientOSX: Where are we at on this PR?  Can we either close and re-open when it's ready, or merge and close?

**Comment by AmbientOSX** [comment]
Its good to merge so.. I'll do so. Need it for new block value visibility changes I'll be working on tomorrow anyway :)

**Pull request merged**